src/docker-extract.c

```diff
@@ -294,6 +301,15 @@ int main(int argc, char **argv) {
     char *rootfs_dir = singularity_registry_get("ROOTFS");
     char *tarfile = NULL;
 
+    // Set UTF8 locale so that libarchive doesn't produce warnings for UTF8
+    // names - en_US.UTF-8 is most likely to be available
+    if( setlocale(LC_ALL, "en_US.UTF-8") == NULL ) {
```

Comment:
Is the locale going to revert back to LC_ALL=C at the conclusion of this function, or will it remain changed until we exit the C code?  If it does not revert, how about setting it back to C when you are done?  There is an example for how to do this safely [here](https://www.gnu.org/software/libc/manual/html_node/Setting-the-Locale.html).

Reply:
Where in main here - so when we drop out, our short-lived docker-extract process is gone, and we're back to the shell script that called it (which will be under whatever locale it was initiated with). The UTF8 won't propagate outwards to anything outside of our current docker-extract instance.

Reply:
Right.  And the two shell scripts that call this are `build.exec` and the docker image handler so we are good.  Thanks.